tests/src/test/scala/akka/kafka/scaladsl/RebalanceExtSpec.scala

```diff
@@ -0,0 +1,427 @@
+      )
+
+      // consumer-1::introduce first consumer with topic-1-1-0 assigned to its SubSource-topic-1-1-0-A
+      val subscription = Subscriptions.topics(topicSet)
+      val sharedKillSwitch1: SharedKillSwitch = KillSwitches.shared(consumerClientId1)
+      val (control1, _) =
+        subscribeAndConsumeMessages(
+          consumerClientId1,
+          perPartitionMessageCount,
+          topicPartitionFutureMap,
+          messageStoreAndAck,
+          consumerSettings1,
+          subscription,
+          topicCount,
+          partitionCount,
+          sharedKillSwitch1,
+          businessFlow1
+        )
+      // consumer-1::SubSource-topic-1-1-0-A:confirm first messadeId=1 is received and committed from message batch (1,2,3)
+      messageStoreAndAck(1)._2.complete(Try(Done))
+      Await.result(messageStoreAndAck(1)._3.future, maxAwait)
+      // consumer-1::SubSource-topic-1-1-0-A:verify messageId=1 is received in the business logic function
+      assert(messageStoreAndAck(1)._1.intValue() == 1)
+
+      // consumer-2::introduce second consumer with stopic-1-1-1 assigned to its SubSource-1-1-1-A
+      val sharedKillSwitch2: SharedKillSwitch = KillSwitches.shared(consumerClientId2)
+      val (control2, _) =
+        subscribeAndConsumeMessages(
+          consumerClientId2,
+          perPartitionMessageCount,
+          topicPartitionFutureMap,
+          messageStoreAndAck,
+          consumerSettings1,
+          subscription,
+          topicCount,
+          partitionCount,
+          sharedKillSwitch2,
+          businessFlow1
+        )
+      // consumer-2::SubSource-topic-1-1-1-A:confirm first messageId=10 is received and committed from batch (10,11,12)
+      messageStoreAndAck(10)._2.complete(Try(Done))
+      Await.result(messageStoreAndAck(10)._3.future, maxAwait)
+      // consumer-2::SubSource-topic-1-1-1-A:verify messageId=10 is received in the business logic function
+      assert(messageStoreAndAck(10)._1.intValue() == 1)
+
+      // consumer-2::define post-abort partition distribution
+      AlpakkaAssignor.clientIdToPartitionMap.set(
+        Map(
+          consumerClientId2 -> Set(t1p0, t1p1)
+        )
+      )
+      // consumer-1::SubSource-topic-1-1-0-A:unblock messageId=2 from batch (1,2,3)
+      messageStoreAndAck(2)._2.complete(Try(Done))
+      // consumer-1::SubSource-topic-1-1-0-A:verify messageId=2 thread is unblocked
+      Await.result(messageStoreAndAck(2)._3.future, maxAwait)
+      // assert(messageStoreAndAck(2)._1.intValue() == 1)
+
+      // consumer-1::SubSource-topic-1-1-0-A:abort at messageId=2 (uncommitted)
+      sharedKillSwitch1.abort(new Throwable(s"abort $consumerClientId1 messageId=2"))
+
+      // consumer-2::after abort two new sub sources serve topic-1-1-0 and topic-1-1-1: SubSource-topic-1-1-0-B and SubSource-topic-1-1-1-B
+
+      // consumer-2::SubSource-topic-1-1-1-A:unblock second message from batch (10,11,12)
+      messageStoreAndAck(11)._2.complete(Try(Done))
```

Comment:
Better return types will make this and other usages more readable.

Reply:
done